src/main/java/sg/ncl/MainController.java

```diff
@@ -2389,50 +2403,56 @@ public String saveExperimentImage(
             @PathVariable String expId,
             @PathVariable String nodeId) throws IOException {
 
-        if (saveImageForm.getImageName().length() < 2) {
+        String REDIRECT_SAVING_IMAGE = "redirect:/experiments/save_image/";
+
+        //checking image name
+        String imageName = saveImageForm.getImageName();
+        if (imageName.length() < 2) {
             log.warn("Save image form has errors {}", saveImageForm);
-            redirectAttributes.addFlashAttribute("message", "Image name too short, minimum 2 characters");
-            return "redirect:/experiments/save_image/" + teamId + "/" + expId + "/" + nodeId;
+            redirectAttributes.addFlashAttribute("message", "Image name is required minimum of 2 characters");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'MESSAGE' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)